src/alignment/pairwise.rs

```diff
@@ -96,37 +95,52 @@ macro_rules! align {
                     let a = $x[$state.j - 1];
 
                     // score for deletion
-                    let d_score = max(
-                        $aligner.S[prev][$state.j] + $aligner.gap_open,
-                        $aligner.D[prev][$state.j] + $aligner.gap_extend
-                    );
+                    let d_open = $aligner.S[prev][$state.j] + $aligner.gap_open;
+                    let d_extend = $aligner.D[prev][$state.j] + $aligner.gap_extend;
+
+                    if d_open > d_extend {
+                        $aligner.D_traceback.subst($state.i, $state.j);
+                        $aligner.D[$state.col][$state.j] = d_open;
+                    } else {
+                        $aligner.D_traceback.del($state.i, $state.j);
+                        $aligner.D[$state.col][$state.j] = d_extend;
+                    }
+
                     // score for insertion
-                    let i_score = max(
-                        $aligner.S[$state.col][$state.j-1] + $aligner.gap_open,
-                        $aligner.I[$state.col][$state.j-1] + $aligner.gap_extend
-                    );
+                    let i_open = $aligner.S[$state.col][$state.j-1] + $aligner.gap_open;
+                    let i_extend = $aligner.I[$state.col][$state.j-1] + $aligner.gap_extend;
+
+                    if i_open > i_extend {
+                        $aligner.I_traceback.subst($state.i, $state.j);
```

Comment:
Maybe I am a bit confused, sorry, but can you explain why opening a gap leads to adding a "substitution" event to I_traceback?